packages/neuron-wallet/src/services/node.ts

```diff
@@ -219,38 +235,58 @@ class NodeService {
     }
   }
 
-  private async verifyNodeVersion() {
-    const network = NetworksService.getInstance().getCurrent()
-    const localNodeInfo = await new RpcService(network.remote).localNodeInfo()
-    const internalNodeVersion = this.getInternalNodeVersion()
-    const [internalMajor, internalMinor] = internalNodeVersion?.split('.') ?? []
-    const [externalMajor, externalMinor] = localNodeInfo.version?.split('.') ?? []
+  private getNeuronCompatibilityCKB(neuronVersion: string) {
+    const appPath = electronApp.isPackaged ? electronApp.getAppPath() : path.join(__dirname, '../../../..')
+    const compatiblePath = path.join(appPath, 'compatible.csv')
+    if (fs.existsSync(compatiblePath)) {
+      try {
+        const neuronCompatibleVersion = neuronVersion.split('.').slice(0, 2).join('.')
+        const content = fs.readFileSync(compatiblePath, 'utf8')?.split('\n')
+        const ckbVersions = content?.[0]?.split(',')?.slice(1)
+        const neuronCompatible = content
+          .find(v => v.startsWith(neuronCompatibleVersion))
+          ?.split(',')
+          ?.slice(1)
+        if (neuronCompatible?.length && ckbVersions?.length) {
+          return neuronCompatible
+            .map((v: 'yes' | 'no', idx) => {
+              if (v === 'yes') return ckbVersions[idx]
+            })
+            .filter((v): v is string => !!v)
+        }
+      } catch (err) {
+        logger.error('App\t: get compatible table failed')
+      }
+    }
+  }
 
-    if (internalMajor !== externalMajor || (externalMajor === '0' && internalMinor !== externalMinor)) {
-      dialog.showMessageBox({
-        type: 'warning',
-        message: t('messageBox.node-version-different.message', { version: internalNodeVersion }),
-      })
+  private getCompatibility(neuronCKBVersion: string, externalCKBVersion: string) {
+    const [internalMajor, internalMinor] = neuronCKBVersion.split('.')?.map(v => +v) ?? []
+    const [externalMajor, externalMinor] = externalCKBVersion.split('.')?.map(v => +v) ?? []
+
+    if (internalMajor === externalMajor && internalMinor === externalMinor) return VerifyCkbVersionResult.Same
+    if (internalMajor < externalMajor || (internalMajor === externalMajor && internalMinor < externalMinor)) {
+      return VerifyCkbVersionResult.ShouldUpdate
     }
+    const supportCkbVersions = this.getNeuronCompatibilityCKB(neuronCKBVersion)
+    if (supportCkbVersions?.every(v => !externalCKBVersion.startsWith(v))) {
+      return VerifyCkbVersionResult.Incompatible
+    }
+    return VerifyCkbVersionResult.Compatible
```

Comment:
What if loading the compatibility result from the compatibility table

Reply:
`getNeuronCompatibilityCKB` loading the compatibility result from the compatibility table

Reply:
My bad, but why not `get compatibility result` first, and check `should update` when it returns `incompatible`. The logic would be quite simple.

Reply:
The logic of locating the table is a bit weird. Usually we locate the value in a table by `row` and `column`, e.g. `result = table[row][column]`. In this case, the logic is 

```
1. columns = table[row]
2. return yes if column in columns
```

A bit hard to read.

Reply:
> My bad, but why not `get compatibility result` first, and check `should update` when it returns `incompatible`. The logic would be quite simple.

I found that `VerifyCkbVersionResult.ShouldUpdate` should return by another field, because `ShouldUpdate` may intersect with `VerifyCkbVersionResult.Incompatible` and `VerifyCkbVersionResult.Compatible`.
So I will change the return type of `verifyExternalCkbNode`. And the `getCompatibility` function will only return true or false.

```typescript
// verifyExternalCkbNode return type
Promise<{
ckbIsCompatible: boolean
withIndexer: boolean
shouldUpdate: boolean
}>
```

Reply:
> > The logic of locating the table is a bit weird. Usually we locate the value in a table by `row` and `column`, e.g. `result = table[row][column]`. In this case, the logic is
> > ```
> > 1. columns = table[row]
> > 2. return yes if column in columns
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > A bit hard to read.
> 
> Sorry, I can not get the point. Finally, we need to verify whether the value is `yes` to `no`

An intuitive way to get compatibility would be `table[neuronVersion][ckbVersion] === yes`. The current implementation is

```
const supportedCkbVersion = table[neuronVersion].filter(compatible => compatible === yes)
const compatible = supportedCkbVersion.includes(ckbVersion)
```